cmd/lotus-bench/simple.go

```diff
@@ -0,0 +1,980 @@
+		if err != nil {
+			return xerrors.Errorf("parse commr: %w", err)
+		}
+
+		rvp, err := sealer.ProveReplicaUpdate1(ctx, sr, oldcommr, commr, commd)
+		if err != nil {
+			return xerrors.Errorf("replica update: %w", err)
+		}
+
+		took := time.Now().Sub(start)
+
+		fmt.Printf("ProveReplicaUpdate1 %s (%s)\n", took, bps(sectorSize, 1, took))
+
+		vpjb, err := json.Marshal(&rvp)
+		if err != nil {
+			return xerrors.Errorf("json marshal vanilla proofs: %w", err)
+		}
+
+		if err := ioutil.WriteFile(cctx.Args().Get(7), vpjb, 0666); err != nil {
+			return xerrors.Errorf("writing vanilla proofs file: %w", err)
+		}
+
+		return nil
+	},
+}
+
+var simpleProveReplicaUpdate2 = &cli.Command{
+	Name: "provereplicaupdate2",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "sector-size",
+			Value: "512MiB",
+			Usage: "size of the sectors in bytes, i.e. 32GiB",
+		},
+		&cli.StringFlag{
+			Name:  "miner-addr",
+			Usage: "pass miner address (only necessary if using existing sectorbuilder)",
+			Value: "t01000",
+		},
+	},
+	ArgsUsage: "[sectorKey] [newSealed] [newUnsealed] [vproofs.json]",
+	Action: func(cctx *cli.Context) error {
+		ctx := cctx.Context
+
+		maddr, err := address.NewFromString(cctx.String("miner-addr"))
+		if err != nil {
+			return err
+		}
+		amid, err := address.IDFromAddress(maddr)
+		if err != nil {
+			return err
+		}
+		mid := abi.ActorID(amid)
+
+		sectorSizeInt, err := units.RAMInBytes(cctx.String("sector-size"))
+		if err != nil {
+			return err
+		}
+		sectorSize := abi.SectorSize(sectorSizeInt)
+
+		pp := benchSectorProvider{}
+		sealer, err := ffiwrapper.New(pp)
+		if err != nil {
+			return err
+		}
+
+		sr := storage.SectorRef{
+			ID: abi.SectorID{
+				Miner:  mid,
+				Number: 1,
+			},
+			ProofType: spt(sectorSize),
+		}
+
+		start := time.Now()
+
+		oldcommr, err := cid.Parse(cctx.Args().Get(0))
+		if err != nil {
+			return xerrors.Errorf("parse commr: %w", err)
+		}
+
+		commr, err := cid.Parse(cctx.Args().Get(1))
+		if err != nil {
+			return xerrors.Errorf("parse commr: %w", err)
+		}
+
+		commd, err := cid.Parse(cctx.Args().Get(2))
+		if err != nil {
+			return xerrors.Errorf("parse commr: %w", err)
+		}
+
+		vpb, err := ioutil.ReadFile(cctx.Args().Get(3))
+		if err != nil {
+			return xerrors.Errorf("reading valilla proof file: %w", err)
+		}
+
+		var vp storage.ReplicaVanillaProofs
+		if err := json.Unmarshal(vpb, &vp); err != nil {
+			return xerrors.Errorf("unmarshalling vanilla proofs: %w", err)
+		}
+
+		p, err := sealer.ProveReplicaUpdate2(ctx, sr, oldcommr, commr, commd, vp)
+		if err != nil {
+			return xerrors.Errorf("prove replica update2: %w", err)
+		}
+
+		took := time.Now().Sub(start)
+
+		fmt.Printf("ProveReplicaUpdate2 %s (%s)\n", took, bps(sectorSize, 1, took))
+		fmt.Println("p:", base64.StdEncoding.EncodeToString(p))
+
+		return nil
+	},
+}
+
+func ParsePieceInfos(cctx *cli.Context, firstArg int) ([]abi.PieceInfo, error) {
+	// supports only one for now
```

Comment:
You should probably make this functions support multiple piece infos, or at least make this a todo so it's more visible.